tests/integration/cmd_list_services.go

```diff
@@ -0,0 +1,67 @@
+package integration
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-developer/odo/tests/helper"
+	"github.com/tidwall/gjson"
+)
+
+var _ = Describe("odo list services tests", func() {
+	var commonVar helper.CommonVar
+	var randomProject string
+
+	BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach(helper.SetupClusterTrue)
+
+		// Ensure that the operators are installed
+		commonVar.CliRunner.EnsureOperatorIsInstalled("service-binding-operator")
+		commonVar.CliRunner.EnsureOperatorIsInstalled("cloud-native-postgresql")
+		Eventually(func() string {
+			out, _ := commonVar.CliRunner.GetBindableKinds()
+			return out
+		}, 120, 3).Should(ContainSubstring("Cluster"))
+		addBindableKind := commonVar.CliRunner.Run("apply", "-f", helper.GetExamplePath("manifests", "bindablekind-instance.yaml"))
+		Expect(addBindableKind.ExitCode()).To(BeEquivalentTo(0))
+		randomProject = helper.CreateRandProject()
+		addBindableKind = commonVar.CliRunner.Run("apply", "-n", randomProject, "-f", helper.GetExamplePath("manifests", "bindablekind-instance.yaml"))
+		Expect(addBindableKind.ExitCode()).To(BeEquivalentTo(0))
+		helper.Cmd("odo", "set", "project", commonVar.Project).ShouldPass()
+	})
+
+	AfterEach(func() {
+		helper.CommonAfterEach(commonVar)
+		helper.Cmd("odo", "delete", "project", randomProject, "-f").ShouldPass()
+	})
+
+	It("should list bindable services in human readable format", func() {
+		// from current namespace
+		out := helper.Cmd("odo", "list", "services", "-n", commonVar.Project).ShouldPass().Out()
+		helper.MatchAllInOutput(out, []string{"cluster-sample", commonVar.Project, "Listing bindable services from namespace"})
+
+		// from all namespaces
+		out = helper.Cmd("odo", "list", "services", "-A").ShouldPass().Out()
+		helper.MatchAllInOutput(out, []string{"cluster-sample", commonVar.Project, randomProject, "Listing bindable services from all namespaces"})
+
+		// fail if -A and -n flags are used together
+		out = helper.Cmd("odo", "list", "services", "-A", "-n", commonVar.Project).ShouldFail().Err()
```

Comment:
Can this be a separate `It` spec instead?

Reply:
I'd prefer to keep it here because a separate `It` spec would mean more time to spin up and tear things down. I wouldn't mind doing it if there was something where we were testing more stuff. Here, it's quite small to put in a separate `It`, IMO.

Reply:
> Yes, I was thinking about having this test in its own container node, which would not have all the setup/teardown code.
> This would help in terms of reporting and also for test parallelization, because I think we are testing independent things here: testing that "-A" and "-n" cannot be used together should be run even if the "odo list services -A" command above is returning the wrong data.

Sounds like a good thing. Is it the `By` directive of ginkgo that helps do this? Can you point me to an example if there's already one in the odo code?

Reply:
[`By`](https://onsi.github.io/ginkgo/#documenting-complex-specs-by) is syntactic sugar that generates "STEP: " in reports, to better document complex specs. I think this could have been used for example to separate the checks against the current namespace and all namespaces.

I was thinking of having those tests under a completely different `Context` or `Describe` container node, like:
```go
var _ = Describe("odo list services tests", func() {

	Context("passing the wrong arguments", func() {
		var commonVar helper.CommonVar

		var _ = BeforeEach(func() {
            // no need to create a project
			commonVar = helper.CommonBeforeEach(helper.SetupClusterFalse)
			helper.Chdir(commonVar.Context)
		})

		var _ = AfterEach(func() {
			helper.CommonAfterEach(commonVar)
		})

		It("should fail if -A and -n flags are used together", func() {
			By("human-readable output", func() {
				out := helper.Cmd("odo", "list", "services", "-A", "-n", "any-namespace").ShouldFail().Err()
				Expect(out).To(ContainSubstring("cannot use --all-namespaces and --namespace flags together"))
			})
			By("JSON output", func() {
				out := helper.Cmd("odo", "list", "services", "-A", "-n", "any-namespace", "-o", "json").ShouldFail().Err()
				Expect(out).To(ContainSubstring("cannot use --all-namespaces and --namespace flags together"))
			})
		})
	})

	Context("passing the right arguments", func() {
      // -- Truncated : the other tests --
```

There should be some examples in odo (like in `cmd_dev_test.go`), but I think they are also nested in other container nodes.